packages/vite/src/node/plugins/worker.ts

```diff
@@ -44,23 +44,24 @@ function saveEmitWorkerAsset(
 // Ensure that only one rollup build is called at the same time to avoid
 // leaking state in plugins between worker builds.
 // TODO: Review if we can parallelize the bundling of workers.
+const workerKey: { key: string } = { key: 'vite:worker' }
 const workerConfigSemaphore = new WeakMap<
-  ResolvedConfig,
+  typeof workerKey,
```

Comment:
After this change, the semaphore would be global and could be replaced by a direct variable to the `Promise<OutputChunk>`. But AFAICS, this shouldn't fix the issue in general, it should make it happen more stable.

I think we need something like: https://github.com/vitejs/vite/pull/14113 for this to work.

Reply:
yeah, we need another solution... closing this